### ISSUE TITLE
Auth.fetchAuthSession throws error after successful Auth.federateToIdentityPool

### ISSUE THREAD

**Issue by Li11leprince** [aws-amplify/amplify-swift#3148]
### Describe the bug

We are trying to Sign in with Apple using federateToIdentityPool method with Apple ID token. The method returns success but after we try to fetch auth session in order to get tokens we get an error: AuthError: Users Federated to Identity Pool do not have User Pool access. We used this as a reference: https://docs.amplify.aws/lib/auth/advanced/q/platform/ios/. The code is following:
```swift
func handleCredentialsCognito(idToken: String) {
        guard
            let plugin = try? Amplify.Auth.getPlugin(for: "awsCognitoAuthPlugin"),
            let authPlugin = plugin as? AWSCognitoAuthPlugin
        else { return }
        Task { @MainActor [weak self] in
            do {
                let result = try await authPlugin.federateToIdentityPool(
                    withProviderToken: idToken,
                    for: .apple
                )
                let session = try await Amplify.Auth.fetchAuthSession() //the error is here
                if let cognitoTokenProvider = session as? AuthCognitoTokensProvider {
                        let tokens = try cognitoTokenProvider.getCognitoTokens().get()
                    }
            } catch let error {
                print(error.localizedDescription)
            }
        }
    }
```

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Register Sign in with Apple in App Store Connect
2. amplify add auth
 Enter your Bundle Identifier for your identity pool:  my.app
 Do you want to add User Pool Groups? No
 Do you want to add an admin queries API? No
 Multifactor authentication (MFA) user login options: OFF
 Email based user registration/forgot password: Enabled (Requires per-user email entry at regi
stration)
 Specify an email verification subject: Your verification code
 Specify an email verification message: Your verification code is {####}
 Do you want to override the default password policy for this User Pool? No
 Specify the app's refresh token expiration period (in days): 30
 Do you want to specify the user attributes this app can read and write? No
 Do you want to enable any of the following capabilities? 
 Do you want to use an OAuth flow? Yes
 What domain name prefix do you want to use? MyApp
 Which redirect signin URIs do you want to edit? 
 Do you want to add redirect signin URIs? No
 Which redirect signout URIs do you want to edit? 
 Do you want to add redirect signout URIs? No
 Select the OAuth scopes enabled for this project. Email, OpenID, Profile
 Select the identity providers you want to configure for your user pool: Sign in with Apple

Enter your Services ID for your OAuth flow:  service.id
 Enter your Team ID for your OAuth flow:  myteam.id
 Enter your Key ID for your OAuth flow:  key.id
 Enter your Private Key for your OAuth flow (entire key without line breaks): private.id

3. import amplify library
4. get Apple ID token with AuthenticationServices
5. pass it in the method above
```

### Expected behavior

We expected fetchAuthSession returning a session from which we could get OAuth tokens

### Amplify Framework Version

2.15.2

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.8.1

### CLI version

12.2.4

### Xcode version

14.3.1

### Is this a regression?

No

### Platforms

iOS

### OS Version

16.0

### Device

simulator iPhone 14

**Comment by Li11leprince** [comment]
In the documentation (https://docs.amplify.aws/lib/auth/advanced/q/platform/ios/) it says that after federated sign in I can retrieve session with fetchAuthSession.
If it doesn't work how can I get the tokens using federateToIdentityPool?
We wanna use social Sign in with Apple ID token that we get using AuthenticationServices. And after get OAuth tokens to handle authorization in our backend

**Comment by Li11leprince** [comment]
The fetchAuthSession didn't succeed but i believe it's doesn't matter anymore if i can not get the tokens from there.
How then I can get the cognito tokens by providing an apple id token?

**Comment by harsh62** [comment]
I would suggest you to go through https://github.com/aws-amplify/amplify-swift/issues/1121 .. 

TLDR: You would need to use HostedUI for that i.e. `signInWithWebUI` API.

**Comment by Li11leprince** [comment]
Oh i don't really wanna use that. We faced a lot of problems when we used web view in sign in flow last time. Is there any chance i can accomplish my goal just by providing apple id token to somethere?

**Comment by harsh62** [comment]
Unfortunately, the only way you can federate into Cognito Pools is using HostedUI. You can track the issue I linked where we will post an update once we more details from Cognito Service.

**Comment by Li11leprince** [comment]
Okey, thank you for your time!

**Comment by harsh62** [comment]
Closing the issue in favor of https://github.com/aws-amplify/amplify-swift/issues/1121.
